app/services/activity-expectations.js

```diff
@@ -33,6 +33,13 @@ export default Service.extend({
   idsToExpectations: activityExpectations,
 
   expectationFor(challenge) {
+    
+    const _combinedExpectations = this.combinedExpectations(challenge)
+
+    return this.expectationsExist(_combinedExpectations) ? this.expectations(this.mergedExpectations(_combinedExpectations)) : noExpectation
+  },
+
+  combinedExpectations(challenge){
```

Comment:
Ahora que hay dos métodos  (merge y combine) me molesta más la diferencia entre ambos y quiero pensar mejor el nombre.  Además, se confunde la idea de la función expectation con la del objeto pavo "config".

- [ ] `expectations(...)` podría llamarse `configToExpectation` (recibe 1 config y devuelve 1 sola expectation)
- [ ] `expectationsName` podría llamarse `expectationsConfigPropertyName`, o, si me permitís para simplificar, directamente pondría `'expectations'` el map de abajo.
- [ ] `combinedExpectations` no combina nada, sólo busca y recolecta así que podría llamarse `allExpectConfigurations`
- [ ] `mergedExpectations` podría llamarse `mergeConfigurations`, que devuelve una configuración.
- [ ] `activityExpectations` podría llamarse `idsToExpectations` y arriba en lugar de poner `idsToExpectations: activityExpectations,` se puede poner `idsToExpectations,`
- [ ] Ahora con esas responsabilidades claras, se ve que la responsabilidad del if para devolver la `noExpectation` debería ser del `configToExpectation` y no de `expectationFor`.
- [ ] Y ahora `expectationFor` queda más claro:
```js
expectationFor(challenge) {
        return configToExpectation(mergeConfigurations(allExpectConfigurations(challenge)))
  },
```

Reply:
compose(configToExpectation, mergeConfigurations, allExpectConfigurations)(challenge) (?

Reply:
Por alguna razón no le gusta mucho el `compose`, está tirando error al ejecutar `configToExpectation`, más en concreto me tira que `this` es `undefined` en esta línea:

https://github.com/Program-AR/pilas-bloques/blob/03e28c4de5ce409baff9dde3f574619a9ea5f24d/app/services/challenge-expectations.js#L61

Todavía no me puse a leer cómo funciona el _compose_, mañana lo miro ¿alguna idea de por qué puede ser?

Reply:
Yo no lo usé todavía. No me complicaría la vida y lo dejaría como está. No es como componer en haskell que queda bonito, acá hasta la estás agregando boilerplate, es medio paja.

Si el `this` es `undefined` supongo que el compose te devuelve una función en la cual tenés que establecerle el contexto y habría que hacer algo como: `compose(...).call(this, challenge)` así establecemos cuál es el `this` en la función.